pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -388,6 +395,9 @@ def _evaluate(
         # log callback metrics
         self.__update_callback_metrics(eval_results, using_eval_result)
 
+        # Write predictions to disk if they're available.
+        predictions.to_disk()
```

Comment:
this should just be an internal function of the prediction object.

Reply:
we can’t wait till the end of epoch to write predictions bc we will accumulate too much memory. 

the write needs to happen at every batch.

Reply:
how do we want to deal with writing to the cache file (w/ `torch.save`) if we want to append to it?